src/game_character.cpp

```diff
@@ -266,6 +266,9 @@ void Game_Character::MoveTypeCustom() {
 					looped_around = true;
 					active_route_index = 0;
 					SetMoveRouteRepeated(true);
+					if (original_index == 0 && looped_around) {
```

Comment:
You set `looped_around` to true directly above. Seems like this check is not needed then.